spec/lang/representation.md

```diff
@@ -440,6 +452,48 @@ fn ensure_else_ub(b: bool, msg: &str) -> Result<()> {
 }
 
 impl<M: Memory> Machine<M> {
+    /// Checks that a pointer is well-formed.
+    fn check_pointer(&self, ptr: Pointer<M::Provenance>, ptr_ty: PtrType) -> Result {
+        ensure_else_ub(ptr.thin_pointer.addr.in_bounds(Unsigned, M::T::PTR_SIZE), "Value::Ptr: pointer out-of-bounds")?;
+
+        // Safe pointer, i.e. references, boxes
+        if let Some(pointee) = ptr_ty.safe_pointee() {
+            let size = pointee.layout.compute_size(ptr.metadata, self.vtable_lookup());
+            let align = pointee.layout.compute_align(ptr.metadata, self.vtable_lookup());
```

Comment:
Can we have `self.compute_size`/`self.compute_align` helper methods for this?

Reply:
Ok, I understood correctly. Currently `self.compute_size == self.size_computer()` (using partial application). Thats what I meant.